homeassistant/components/otbr/config_flow.py

```diff
@@ -101,27 +108,66 @@ async def _connect_and_set_dataset(self, otbr_url: str) -> None:
                 )
             await api.set_enabled(True)
 
+    async def _is_extended_address_configured(self, extended_address: bytes) -> bool:
+        """Return True if another config entry's OTBR has the same extended address."""
+        if DATA_OTBR not in self.hass.data:
+            _LOGGER.debug("No OTBR")
+            return False
+        for data in self.hass.data[DATA_OTBR].values():
+            try:
+                other_extended_address = await data.get_extended_address()
+            except HomeAssistantError:
+                _LOGGER.debug(
+                    "Could not read extended address from %s", data.url, exc_info=True
+                )
+                continue
+            _LOGGER.debug(
+                "extended address for url %s: %s",
+                data.url,
+                other_extended_address.hex(),
+            )
+            if extended_address == other_extended_address:
+                return True
+        return False
+
+    async def _connect_and_configure_router(self, otbr_url: str) -> bytes:
+        """Connect to the router and configure it if needed.
+
+        Will raise if the router's extended address is in use by another confi entry.
+        Returns the router's extended address.
+        """
+        api = python_otbr_api.OTBR(otbr_url, async_get_clientsession(self.hass), 10)
+        extended_address = await self._get_extended_address(api)
+        _LOGGER.debug("extended_address: %s", extended_address.hex())
+
+        if await self._is_extended_address_configured(extended_address):
```

Comment:
Should we maybe also check the URL string itself? :thinking: 

Hm, on second thought, we only can add an online OTBR, and if the other URL is online, we'll already detect that. So not really necessary.

Reply:
Btw, if we change to `id` (border agent ID) as unique ID, I think we should check both, the extended address and the unique id. Just to avoid any strange behavior in case of a messed up setup (since we rely on extended addresses on the frontend.

Reply:
> if we change to id (border agent ID) as unique ID, I think we should check both, the extended address and the unique id. Just to avoid any strange behavior in case of a messed up setup

Let's add this if we see issues instead of complicating the code now

Reply:
Technically, it is possible to generate such an issue since the extaddr can be manually set.

However, practically it is very unrealistic, as both the border agent id and the extended address is randomly generated. So this would most likely be a "fabricated" case.

So if this over-complicates code, I am fine not doing it now :+1: